src/sortStudents.ts

```diff
@@ -1,16 +1,74 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copy = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  const average = (numbers: number[]): number => {
+    return numbers.reduce((a, b) => a + b, 0) / numbers.length;
+  };
+
+  copy.sort((studentOne: Student, studentTwo: Student) => {
+    switch (sortBy) {
+      case SortType.Name:
+        if (order === 'asc') {
+          return studentOne.name.localeCompare(studentTwo.name);
+        }
+
+        return studentTwo.name.localeCompare(studentOne.name);
+
+      case SortType.Surname:
+        if (order === 'asc') {
+          return studentOne.surname.localeCompare(studentTwo.surname);
+        }
+
+        return studentTwo.surname.localeCompare(studentOne.surname);
+
```

Comment:
Combine similar sort logic to one case
Use computed properties keys
```suggestion
      case SortType.Name:
            case SortType.Surname:
        if (order === 'asc') {
          return studentOne[sortBy].localeCompare(studentTwo[sortBy]);
        }

        return studentTwo[sortBy].localeCompare(studentOne[sortBy];

```